### PR TITLE
refactor: configuration to use Platform SDK across all project

### LINKED ISSUE #33: Add Configuration from `application.yaml` properties file with default values.
### Problem

Currently Helidon is not picking up properties from `properties.yaml` file, we need to include this file from resources and override them with `application.yaml` values if specified. 

Properties on `application.yaml` should contain the default values.

### Solution

Figure out a loading mechanism that allows us to load default values from yaml, and override only the ones explicitly defined as ENVIRONMENT variables

### Alternatives

_No response_

### LINKED ISSUE #98: Helidon doesn't resolve configuration properly
Currently application is trying to use application.yaml for configuration, this results in error when trying to start the block-node.

```
WARNING: Configuration file application.yaml is on classpath, yet there is no parser configured for it
Exception in thread "main" io.helidon.config.MissingValueException: Requested value for configuration key 'blocknode.storage.root.path' is not present in the configuration.
	at io.helidon.config@4.0.11/io.helidon.config.MissingValueException.create(MissingValueException.java:44)
	at io.helidon.config@4.0.11/io.helidon.config.ConfigValue.lambda$get$0(ConfigValue.java:93)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at io.helidon.config@4.0.11/io.helidon.config.ConfigValue.get(ConfigValue.java:93)
	at com.hedera.block.server/com.hedera.block.server.persistence.storage.write.BlockAsDirWriterBuilder.<init>(BlockAsDirWriterBuilder.java:55)
	at com.hedera.block.server/com.hedera.block.server.persistence.storage.write.BlockAsDirWriterBuilder.newBuilder(BlockAsDirWriterBuilder.java:73)
	at com.hedera.block.server/com.hedera.block.server.Server.main(Server.java:72)
```

We'd need to fix the configurator to start using `app.properties` files for configuration variables.

### LINKED ISSUE #99: Refactor Project Configuration to use PlatformSDK instead of Helidon.
### Problem

Currently we are using Platform SDK configuration API for Metrics.
But we are using Helidon Configuration for features.

### Solution

We need to remove the properties from Helidon and refactor the different classes with features to use the Platform SDK Metrics and approach.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
- Created the following config classes
  - ConsumerConfig
  - PersistenceStorageConfig
- Refactored Classes to use those config instead of Helidon config
- Simplified Classes that already had BlockNodeContext to use config from there directly
- Refactored and Simplified all related Unit Tests
- 

**Related issue(s)**:

Fixes #98 
Fixes #99 
Fixes #33 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by mattp-swirldslabs — COMMENTED** [review]
Looks good. Ping me to review again when you're ready

File: server/src/main/java/com/hedera/block/server/persistence/storage/write/BlockAsDirWriter.java [hunk 1]
```diff
@@ -57,35 +56,31 @@ class BlockAsDirWriter implements BlockWriter<BlockItem> {
      * Constructor for the BlockAsDirWriter class. It initializes the BlockAsDirWriter with the
      * given key, config, block remover, and file permissions.
      *
-     * @param key the key to use to retrieve the block node root path from the config
-     * @param config the config to use to retrieve the block node root path
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply server/src/main/java/com/hedera/block/server/persistence/storage/write/BlockAsDirWriter.java:61]: ✅ Done!

File: server/src/test/java/com/hedera/block/server/config/TestConfigBuilder.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+     *
+     * @return the created configuration
+     */
+    @NonNull
+    public Configuration getOrCreateConfig() {
+        try (final Locked ignore = configLock.lock()) {
+            if (configuration == null) {
+                configuration = builder.build();
+                ConfigurationHolder.getInstance().setConfiguration(configuration);
+            }
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply server/src/test/java/com/hedera/block/server/config/TestConfigBuilder.java:174]: Yes, However Services is still using it just for tests, and if and when they remove it, they most provide with an alternative. so for now I think we are safe to use it only for tests.

File: server/src/main/java/com/hedera/block/server/config/BlockNodeConfigExtension.java [hunk 3]
```diff
@@ -28,6 +30,11 @@
 @AutoService(ConfigurationExtension.class)
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply server/src/main/java/com/hedera/block/server/config/BlockNodeConfigExtension.java:30]: I think it should be fixed. let me know.

**Review by mattp-swirldslabs — APPROVED** [review]
Looks good!

**Pull request merged**